### PR TITLE
React: Cherry-Pick Commit #3

### PR DESCRIPTION
**Pull request by jnaviask**
## Link to Issue
Closes: #3225

## Description of Changes
- Gets react branch up to date with master, besides root_id changes (1e1d3654e628e50f1f822c0a9296ffcadbb0092d and b107d5bc3fadd03e0f5c0b95d4e09f3da62d55c6)

## Test Plan
- Verified key changes by clicking around.
- Profile view page broke: http://localhost:8080/profile/id/14301 -- may need to debug this before merging.

## Other Considerations

- Looks like the user component still has some weirdness, but we can revert those changes if needed. Should do some more thorough QA there.

### CODE REVIEW

File: packages/commonwealth/client/scripts/navigation/customDomainRoutes.tsx [hunk 1]
```diff
@@ -424,8 +424,8 @@ const customDomainRoutes = () => {
       element={<Navigate to="/profile/id/:profileId" />}
     />,
     <Route
-      path="/:scope/profile/id/:profileId/edit"
-      element={<Navigate to="/profile/id/:profileId/edit" />}
+      path="/:scope/profile/edit"
+      element={<Navigate to="/profile/edit" />}
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/navigation/customDomainRoutes.tsx:428]: these 3 routes has to be added/modified in customDomainRoutes as well

Comment (jnaviask) [reply packages/commonwealth/client/scripts/navigation/customDomainRoutes.tsx:428]: The routes are in both custom and common DomainRoutes -- I don't see what needs to be done?

Comment (masvelio) [reply packages/commonwealth/client/scripts/navigation/customDomainRoutes.tsx:428]: thats my bad!

**Comment by jnaviask** [comment]
Manually merged.

**Pull request merged**